Sources/Redirection/RedirectCallbackHandler.swift

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2022 Payoneer Germany GmbH
+// https://www.payoneer.com
+//
+// This file is open source and available under the MIT license.
+// See the LICENSE file for more information.
+
+import Foundation
+import Networking
+
+class RedirectCallbackHandler {
+    private let openAppWithURLNotificationName: NSNotification.Name
+    private var completionBlock: ((Result<OperationResult, Error>) -> Void)?
+    private var receivePaymentNotificationToken: NSObjectProtocol?
+
+    init(openAppWithURLNotificationName: NSNotification.Name) {
+        self.openAppWithURLNotificationName = openAppWithURLNotificationName
+    }
+
+    func addObserver(completion: ((Result<OperationResult, Error>) -> Void)?) {
+        // Remove observers to avoid duplicate completion calls if `addObservers()` were accidentally called twice
+        removeObserver()
+
+        self.completionBlock = completion
+
+        // Received payment result notification
+        receivePaymentNotificationToken = NotificationCenter.default.addObserver(forName: openAppWithURLNotificationName, object: nil, queue: .main) { [removeObserver] notification in
+            guard let url = notification.userInfo?["url"] as? URL else { return }
+            self.handle(receivedURL: url)
+            removeObserver()
+        }
```

Comment:
Both `[removeObserver]` and `self` in line 28 are a strong captures. Best to use `[weak self]` instead:

```suggestion
        receivePaymentNotificationToken = NotificationCenter.default.addObserver(forName: openAppWithURLNotificationName, object: nil, queue: .main) { [weak self] notification in
            guard let url = notification.userInfo?["url"] as? URL else { return }
            self?.handle(receivedURL: url)
            self?.removeObserver()
        }
```